From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david@flypig.co.uk>
Date: Wed, 5 Aug 2026 11:24:54 +0200
Subject: Allow file scheme when loading OpenSearch providers

Allow packaged Sailfish OpenSearch XML providers to load from disk.

The scheme validation moved to OpenSearchLoader in ESR140.
---
 toolkit/components/search/OpenSearchLoader.sys.mjs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/toolkit/components/search/OpenSearchLoader.sys.mjs b/toolkit/components/search/OpenSearchLoader.sys.mjs
index 77b5d333be0d..76c75e60ffdb 100644
--- a/toolkit/components/search/OpenSearchLoader.sys.mjs
+++ b/toolkit/components/search/OpenSearchLoader.sys.mjs
@@ -50,7 +50,7 @@ export async function loadAndParseOpenSearchEngine(
   if (!sourceURI) {
     throw new TypeError("No URI");
   }
-  if (!/^https?$/i.test(sourceURI.scheme)) {
+  if (!/^(?:https?|file)$/i.test(sourceURI.scheme)) {
     throw new TypeError(
       "Unsupported URI scheme passed to SearchEngine constructor"
     );
